From 7f4e43a22117fbeb6a2adbd5cd36e9c43fc60bc4 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 22 Jan 2021 11:15:17 -0600
Subject: [PATCH] fix mikrobus_id.c

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/misc/mikrobus/mikrobus_id.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/mikrobus/mikrobus_id.c b/drivers/misc/mikrobus/mikrobus_id.c
index e891645f745c..183de02cd74f 100644
--- a/drivers/misc/mikrobus/mikrobus_id.c
+++ b/drivers/misc/mikrobus/mikrobus_id.c
@@ -14,7 +14,7 @@
 #include <linux/w1.h>
 #include <linux/nvmem-provider.h>
 
-#include "mikrobus_core.h"
+#include <linux/mikrobus.h>
 
 #define W1_EEPROM_MIKROBUS_ID	0xAC
 
@@ -204,4 +204,4 @@ module_w1_family(w1_family_mikrobus_id);
 MODULE_AUTHOR("Vaishnav M A <vaishnav@beagleboard.org>");
 MODULE_DESCRIPTION("w1 family ac driver for mikroBUS ID EEPROM");
 MODULE_LICENSE("GPL");
-MODULE_ALIAS("w1-family-" __stringify(W1_EEPROM_MIKROBUS_ID));
\ No newline at end of file
+MODULE_ALIAS("w1-family-" __stringify(W1_EEPROM_MIKROBUS_ID));
-- 
2.29.2

