From 69c32aa699560095fce7b4265bbf3cc8eb10a140 Mon Sep 17 00:00:00 2001
From: anithamolbenny <anithamolbenny101@gmail.com>
Date: Fri, 22 Aug 2014 08:45:05 +0530
Subject: [PATCH] Fix for a part of video got flipped from bottom to top

Signed-off-by: anithamolbenny <anithamolbenny101@gmail.com>
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/media/i2c/soc_camera/mt9m114.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/soc_camera/mt9m114.c b/drivers/media/i2c/soc_camera/mt9m114.c
index 9df2119..2c25b7f 100644
--- a/drivers/media/i2c/soc_camera/mt9m114.c
+++ b/drivers/media/i2c/soc_camera/mt9m114.c
@@ -418,7 +418,7 @@ static const struct mt9m114_reg mt9m114_regs_720p[] = {
 	{ MT9M114_CAM_SENSOR_CFG_ROW_SPEED,              0x0001, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MIN,    0x00A0, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MAX,    0x03F7, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_FRAME_LENGTH_LINES,     0x02D5, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_FRAME_LENGTH_LINES,     0x03EE, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_LINE_LENGTH_PCK,        0x060E, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_FINE_CORRECTION,        0x0060, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_CPIPE_LAST_ROW,         0x02D3, 2 },
-- 
2.1.0

