From a9e20f70c6b5fa5b14441a80bf3e074a1aeeb3bf Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 27 Aug 2026 11:48:35 -0500
Subject: [PATCH 3/3] uio/uio_pruss: update pruss_remove for later then 6.6.x

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/uio/uio_pruss.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/uio/uio_pruss.c b/drivers/uio/uio_pruss.c
index 2edf68535644..7b3efbeae588 100644
--- a/drivers/uio/uio_pruss.c
+++ b/drivers/uio/uio_pruss.c
@@ -278,12 +278,11 @@ static int pruss_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int pruss_remove(struct platform_device *dev)
+static void pruss_remove(struct platform_device *dev)
 {
 	struct uio_pruss_dev *gdev = platform_get_drvdata(dev);
 
 	pruss_cleanup(&dev->dev, gdev);
-	return 0;
 }
 
 static const struct of_device_id pruss_dt_ids[] = {
-- 
2.55.0

