From df85c25f3e10dea481ce11fb1c25ac7124193fe9 Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Wed, 28 Jun 2017 22:36:10 +0200
Subject: [PATCH 4/5] uio_pruss: balance clk_enable() with clk_disable()

Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>
---
 drivers/uio/uio_pruss.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/uio/uio_pruss.c b/drivers/uio/uio_pruss.c
index f8072e95b424..9b468d002c4c 100644
--- a/drivers/uio/uio_pruss.c
+++ b/drivers/uio/uio_pruss.c
@@ -126,7 +126,10 @@ static void pruss_cleanup(struct device *dev, struct uio_pruss_dev *gdev)
 			      sram_pool_sz);
 #endif
 	kfree(gdev->info);
-	clk_put(gdev->pruss_clk);
+	if (gdev->pruss_clk) {
+		clk_disable(gdev->pruss_clk);
+		clk_put(gdev->pruss_clk);
+	}
 	kfree(gdev);
 }
 
-- 
2.11.0

