From 826710362a2a1d314d24df0611ed2d8614183298 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 24 Apr 2026 17:29:26 -0500
Subject: [PATCH] backports: remoteproc: from: linux.git

Reference: 12.00.00.07
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/remoteproc/pru_rproc.c | 273 +++++++++++++++++++++++++++++++--
 1 file changed, 263 insertions(+), 10 deletions(-)

diff --git a/drivers/remoteproc/pru_rproc.c b/drivers/remoteproc/pru_rproc.c
index 5e3eb7b86a0e..79d992daed49 100644
--- a/drivers/remoteproc/pru_rproc.c
+++ b/drivers/remoteproc/pru_rproc.c
@@ -14,6 +14,10 @@
 
 #include <linux/bitops.h>
 #include <linux/debugfs.h>
+#include <linux/delay.h>
+#include <linux/io.h>
+#include <linux/iopoll.h>
+#include <linux/interrupt.h>
 #include <linux/irqdomain.h>
 #include <linux/module.h>
 #include <linux/of.h>
@@ -67,6 +71,8 @@
 
 #define MAX_PRU_SYS_EVENTS 160
 
+static const char *pru_names[PRU_TYPE_MAX] = { "PRU", "RTU", "Tx_PRU" };
+
 /**
  * enum pru_iomem - PRU core memory/register range identifiers
  *
@@ -104,9 +110,11 @@ struct pru_private_data {
  * @lock: mutex to protect client usage
  * @fw_name: name of firmware image used during loading
  * @mapped_irq: virtual interrupt numbers of created fw specific mapping
+ * @rsc_tbl: Resource table
  * @pru_interrupt_map: pointer to interrupt mapping description (firmware)
  * @pru_interrupt_map_sz: pru_interrupt_map size
  * @rmw_lock: lock for read, modify, write operations on registers
+ * @irq_vring: IRQ number to use for processing vring buffers
  * @dbg_single_step: debug state variable to set PRU into single step mode
  * @dbg_continuous: debug state variable to restore PRU execution mode
  * @evt_count: number of mapped events
@@ -123,9 +131,11 @@ struct pru_rproc {
 	struct mutex lock;
 	const char *fw_name;
 	unsigned int *mapped_irq;
+	struct resource_table *rsc_tbl;
 	struct pru_irq_rsc *pru_interrupt_map;
 	size_t pru_interrupt_map_sz;
-	spinlock_t rmw_lock;
+	spinlock_t rmw_lock; /* register access lock */
+	int irq_vring;
 	u32 dbg_single_step;
 	u32 dbg_continuous;
 	u8 evt_count;
@@ -503,6 +513,76 @@ static void pru_dispose_irq_mapping(struct pru_rproc *pru)
 	pru->mapped_irq = NULL;
 }
 
+/*
+ * pru_rproc_vring_interrupt() - interrupt handler for processing vrings
+ * @irq: irq number associated with the PRU event MPU is listening on
+ * @data: interrupt handler data, will be a PRU rproc structure
+ *
+ * This handler is used by the PRU remoteproc driver when using PRU system
+ * events for processing the virtqueues. Unlike the mailbox IP, there is
+ * no payload associated with an interrupt, so either a unique event is
+ * used for each virtqueue kick, or both virtqueues are processed on a
+ * single event. The latter is chosen to conserve the usable PRU system
+ * events.
+ */
+static irqreturn_t pru_rproc_vring_interrupt(int irq, void *data)
+{
+	struct pru_rproc *pru = data;
+
+	dev_dbg(&pru->rproc->dev, "got vring irq\n");
+
+	/* process incoming buffers on both the Rx and Tx vrings */
+	rproc_vq_interrupt(pru->rproc, 0);
+	rproc_vq_interrupt(pru->rproc, 1);
+
+	return IRQ_HANDLED;
+}
+
+/* Kick a virtqueue. */
+static void pru_rproc_kick(struct rproc *rproc, int vq_id)
+{
+	struct device *dev = &rproc->dev;
+	struct pru_rproc *pru = rproc->priv;
+	int ret;
+
+	if (list_empty(&pru->rproc->rvdevs))
+		return;
+
+	dev_dbg(dev, "kicking vqid %d on %s%d\n", vq_id,
+		pru_names[pru->data->type], pru->id);
+
+	ret = irq_set_irqchip_state(pru->mapped_irq[0], IRQCHIP_STATE_PENDING, true);
+	if (ret < 0)
+		dev_err(dev, "pruss_intc_trigger failed: %d\n", ret);
+}
+
+/* Register vring irq handler if needed. */
+static int pru_vring_interrupt_setup(struct rproc *rproc)
+{
+	struct device *dev = &rproc->dev;
+	struct pru_rproc *pru = rproc->priv;
+	struct platform_device *pdev = to_platform_device(pru->dev);
+	int ret;
+
+	if (list_empty(&pru->rproc->rvdevs))
+		return 0;
+
+	/* get vring interrupts for supporting virtio rpmsg */
+	pru->irq_vring = platform_get_irq_byname(pdev, "vring");
+	if (pru->irq_vring <= 0)
+		return dev_err_probe(dev, pru->irq_vring, "unable to get vring interrupt\n");
+
+	ret = request_threaded_irq(pru->irq_vring, NULL,
+				   pru_rproc_vring_interrupt, IRQF_ONESHOT,
+				   dev_name(dev), pru);
+	if (ret) {
+		dev_err(dev, "failed to register vring irq handler: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
 /*
  * Parse the custom PRU interrupt map resource and configure the INTC
  * appropriately.
@@ -594,16 +674,47 @@ static int pru_handle_intrmap(struct rproc *rproc)
 	return ret;
 }
 
+/*
+ * Attach to a running remote processor (IPC-only mode)
+ */
+static int pru_rproc_attach(struct rproc *rproc)
+{
+	struct device *dev = &rproc->dev;
+	struct pru_rproc *pru = rproc->priv;
+	int ret = 0;
+	u32 val;
+
+	dev_dbg(dev, "attach\n");
+
+	/*
+	 * Configure PRU Vring interrupt. Even if this configuration fails,
+	 * let the PRU continue execution without IPC.
+	 */
+	ret = pru_vring_interrupt_setup(rproc);
+	if (ret && pru->mapped_irq)
+		pru_dispose_irq_mapping(pru);
+
+	/* pru interrupts already configured. clear the cached map */
+	pru->pru_interrupt_map = NULL;
+	pru->pru_interrupt_map_sz = 0;
+
+	/* Release/enable the PRU execution */
+	val = pru_control_read_reg(pru, PRU_CTRL_CTRL);
+	val |= CTRL_CTRL_EN;
+	pru_control_write_reg(pru, PRU_CTRL_CTRL, val);
+
+	return ret;
+}
+
 static int pru_rproc_start(struct rproc *rproc)
 {
 	struct device *dev = &rproc->dev;
 	struct pru_rproc *pru = rproc->priv;
-	const char *names[PRU_TYPE_MAX] = { "PRU", "RTU", "Tx_PRU" };
 	u32 val;
 	int ret;
 
 	dev_dbg(dev, "starting %s%d: entry-point = 0x%llx\n",
-		names[pru->data->type], pru->id, (rproc->bootaddr >> 2));
+		pru_names[pru->data->type], pru->id, (rproc->bootaddr >> 2));
 
 	ret = pru_handle_intrmap(rproc);
 	/*
@@ -615,28 +726,45 @@ static int pru_rproc_start(struct rproc *rproc)
 	if (ret)
 		return ret;
 
+	ret = pru_vring_interrupt_setup(rproc);
+	if (ret)
+		goto fail;
+
 	val = CTRL_CTRL_EN | ((rproc->bootaddr >> 2) << 16);
 	pru_control_write_reg(pru, PRU_CTRL_CTRL, val);
 
 	return 0;
+
+fail:
+	/* dispose irq mapping - new firmware can provide new mapping */
+	if (pru->mapped_irq)
+		pru_dispose_irq_mapping(pru);
+
+	return ret;
 }
 
 static int pru_rproc_stop(struct rproc *rproc)
 {
 	struct device *dev = &rproc->dev;
 	struct pru_rproc *pru = rproc->priv;
-	const char *names[PRU_TYPE_MAX] = { "PRU", "RTU", "Tx_PRU" };
 	u32 val;
 
-	dev_dbg(dev, "stopping %s%d\n", names[pru->data->type], pru->id);
+	dev_dbg(dev, "stopping %s%d\n", pru_names[pru->data->type], pru->id);
 
 	val = pru_control_read_reg(pru, PRU_CTRL_CTRL);
 	val &= ~CTRL_CTRL_EN;
 	pru_control_write_reg(pru, PRU_CTRL_CTRL, val);
 
+	if (!list_empty(&pru->rproc->rvdevs) && pru->irq_vring > 0)
+		free_irq(pru->irq_vring, pru);
+
 	/* dispose irq mapping - new firmware can provide new mapping */
 	pru_dispose_irq_mapping(pru);
 
+	/* dispose vring mapping as well */
+	if (pru->irq_vring > 0)
+		irq_dispose_mapping(pru->irq_vring);
+
 	return 0;
 }
 
@@ -750,11 +878,6 @@ static void *pru_da_to_va(struct rproc *rproc, u64 da, size_t len, bool is_iram)
 	return va;
 }
 
-static struct rproc_ops pru_rproc_ops = {
-	.start		= pru_rproc_start,
-	.stop		= pru_rproc_stop,
-	.da_to_va	= pru_rproc_da_to_va,
-};
 
 /*
  * Custom memory copy implementation for ICSSG PRU/RTU/Tx_PRU Cores
@@ -956,6 +1079,112 @@ static int pru_rproc_parse_fw(struct rproc *rproc, const struct firmware *fw)
 	return 0;
 }
 
+/*
+ * This function gets called in IPC only mode. Resource table pointer
+ * is retrieved during 'prepare' and cached locally.
+ */
+static struct resource_table *pru_get_loaded_rsc_table(struct rproc *rproc,
+							size_t *rsc_table_sz)
+{
+	struct pru_rproc *pru = rproc->priv;
+
+	*rsc_table_sz = rproc->table_sz;
+	return pru->rsc_tbl;
+}
+/*
+ * is_pru_running - local utility function to check PRU status
+ * @pru: PRU device pointer used for checking core status
+ */
+static u32 is_pru_running(struct pru_rproc *pru)
+{
+	u32 val;
+
+	val = pru_control_read_reg(pru, PRU_CTRL_CTRL);
+	return (val & CTRL_CTRL_RUNSTATE);
+}
+
+/*
+ * For PRU cores, prepare function is used in IPC only mode. Remote proc
+ * core driver calls 'prepare' to facilitate enabling the configuraiton
+ * needed before attaching to the remote processor. In prepare the PRU
+ * is paused from fetching instruction stream, so that the vring and
+ * interrupts are configured before attaching to the PRU unit.
+ */
+static int pru_rproc_prepare(struct rproc *rproc)
+{
+	struct device *dev = &rproc->dev;
+	struct pru_rproc *pru = rproc->priv;
+	int ret = 0;
+	u32 stat = 0;
+	u32 val;
+	const struct firmware *firmware_p;
+
+	/*
+	 * For PRU cores, prepare is used only in IPC mode to pause and
+	 * configure vrings.
+	 */
+	if (rproc->state != RPROC_DETACHED)
+		return ret;
+
+	if (!rproc->firmware) {
+		dev_err(dev, "PRU FW needed to get PRU resource table\n");
+		return -EINVAL;
+	}
+
+	ret = request_firmware(&firmware_p, rproc->firmware, dev);
+	if (ret < 0) {
+		dev_err(dev, "request_firmware failed: %d\n", ret);
+		return ret;
+	}
+
+	/* Pause the PRU execution, while vring buffers are configured */
+	val = pru_control_read_reg(pru, PRU_CTRL_CTRL);
+	val &= ~CTRL_CTRL_EN;
+	pru_control_write_reg(pru, PRU_CTRL_CTRL, val);
+
+	ret = readx_poll_timeout(is_pru_running, pru, stat, !stat, 200, 2000);
+	if (ret) {
+		dev_err(dev, "timeout waiting for PRU to halt\n");
+		return ret;
+	}
+
+	/* parse the FW for INTR maps and resource table */
+	pru_rproc_parse_fw(rproc, firmware_p);
+
+	/* configure the interrupt map */
+	pru_handle_intrmap(rproc);
+
+	/* cache resource table pointer */
+	pru->rsc_tbl = rproc_elf_find_loaded_rsc_table(rproc, firmware_p);
+
+	release_firmware(firmware_p);
+
+	return ret;
+}
+
+/*
+ * Detach from a running remote processor (IPC-only mode)
+ * This callback is for any cleanup operations on detach for later
+ * state changes like re-attach. PRU IRQ mappings are removed here.
+ */
+static int pru_rproc_detach(struct rproc *rproc)
+{
+	struct pru_rproc *pru = rproc->priv;
+
+	if (!list_empty(&pru->rproc->rvdevs) && pru->irq_vring > 0)
+		free_irq(pru->irq_vring, pru);
+
+	pru_dispose_irq_mapping(pru);
+
+	/* dispose vring mapping as well */
+	if (pru->irq_vring > 0)
+		irq_dispose_mapping(pru->irq_vring);
+
+	pru->rsc_tbl = NULL;
+
+	return 0;
+}
+
 /*
  * Compute PRU id based on the IRAM addresses. The PRU IRAMs are
  * always at a particular offset within the PRUSS address space.
@@ -986,6 +1215,17 @@ static int pru_rproc_set_id(struct pru_rproc *pru)
 	return ret;
 }
 
+static struct rproc_ops pru_rproc_ops = {
+	.start		= pru_rproc_start,
+	.stop		= pru_rproc_stop,
+	.kick		= pru_rproc_kick,
+	.da_to_va	= pru_rproc_da_to_va,
+	.prepare	= pru_rproc_prepare,
+	.attach		= pru_rproc_attach,
+	.detach		= pru_rproc_detach,
+	.get_loaded_rsc_table	= pru_get_loaded_rsc_table,
+};
+
 static int pru_rproc_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
@@ -998,6 +1238,7 @@ static int pru_rproc_probe(struct platform_device *pdev)
 	int i, ret;
 	const struct pru_private_data *data;
 	const char *mem_names[PRU_IOMEM_MAX] = { "iram", "control", "debug" };
+	u32 val;
 
 	data = of_device_get_match_data(&pdev->dev);
 	if (!data)
@@ -1067,6 +1308,18 @@ static int pru_rproc_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, rproc);
 
+	val = pru_control_read_reg(pru, PRU_CTRL_CTRL);
+	if (val & CTRL_CTRL_EN) {
+		/*
+		 * Configure for IPC only mode. 'stop' and 'start' callbacks
+		 * are not allowed in this mode.
+		 */
+		dev_dbg(dev, "Configure PRU IPC only mode.\n");
+		rproc->state = RPROC_DETACHED;
+		rproc->ops->stop = NULL;
+		rproc->ops->start = NULL;
+	}
+
 	ret = devm_rproc_add(dev, pru->rproc);
 	if (ret) {
 		dev_err(dev, "rproc_add failed: %d\n", ret);
-- 
2.53.0

