From 10cc9a2823660a17b917840cc09ab5ad4970d9d1 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 22 Mar 2021 15:31:19 -0500
Subject: [PATCH] backports: bindeb-pkg: from: linux.git

Reference: v5.2.21
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/package/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/package/Makefile b/scripts/package/Makefile
index fd854439de0f..ca7f46b562a4 100644
--- a/scripts/package/Makefile
+++ b/scripts/package/Makefile
@@ -1,3 +1,4 @@
+# SPDX-License-Identifier: GPL-2.0-only
 # Makefile for the different targets used to generate full packages of a kernel
 # It uses the generic clean infrastructure of kbuild
 
-- 
2.30.2

