From 86d7aadcf8f78f3d37e9c88361da48ba9214ebcb Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 28 Jan 2021 18:09:05 -0600
Subject: [PATCH] backports: bindeb-pkg: from: linux.git

Reference: v5.2.21
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/package/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/package/Makefile b/scripts/package/Makefile
index fd854439de0f..ca7f46b562a4 100644
--- a/scripts/package/Makefile
+++ b/scripts/package/Makefile
@@ -1,3 +1,4 @@
+# SPDX-License-Identifier: GPL-2.0-only
 # Makefile for the different targets used to generate full packages of a kernel
 # It uses the generic clean infrastructure of kbuild
 
-- 
2.29.2

