From 253ca9bf7c9c58dc8f3fbf5230bffb39cab46dd8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 20 Mar 2023 17:55:30 -0500
Subject: [PATCH 2/2] builddeb: Install our dtbs under /boot/dtbs/$version

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/package/builddeb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index c5ae57167d7c..7498d732728b 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -60,6 +60,7 @@ install_linux_image () {
 	# Only some architectures with OF support have this target
 	if is_enabled CONFIG_OF_EARLY_FLATTREE && [ -d "${srctree}/arch/${SRCARCH}/boot/dts" ]; then
 		${MAKE} -f ${srctree}/Makefile INSTALL_DTBS_PATH="${pdir}/usr/lib/linux-image-${KERNELRELEASE}" dtbs_install
+		${MAKE} -f ${srctree}/Makefile INSTALL_DTBS_PATH="${pdir}/boot/dtbs/${KERNELRELEASE}" dtbs_install
 	fi
 
 	if is_enabled CONFIG_MODULES; then
-- 
2.39.2

