From 8780b8ca0b4bd888ad64f9b75b30636fbab70dfe Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 16 Jun 2014 14:17:40 -0500
Subject: [PATCH 2/2] capes: Makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 5986ff6..b9da1aa 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -288,7 +288,13 @@ dtb-$(CONFIG_ARCH_OMAP3) += am3517-craneboard.dtb \
 	omap3-zoom3.dtb
 dtb-$(CONFIG_SOC_AM33XX) += am335x-base0033.dtb \
 	am335x-bone.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO4.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO4.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-nano.dtb
-- 
2.0.0

