From a5ee140663b9413bb487803dfaa12245ba5b4b7d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 28 Aug 2019 14:46:51 -0500
Subject: [PATCH] fix rt

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 kernel/time/hrtimer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/time/hrtimer.c b/kernel/time/hrtimer.c
index 5eb45a868de9..37a54aefd080 100644
--- a/kernel/time/hrtimer.c
+++ b/kernel/time/hrtimer.c
@@ -934,7 +934,11 @@ void hrtimer_grab_expiry_lock(const struct hrtimer *timer)
 {
 	struct hrtimer_clock_base *base = READ_ONCE(timer->base);
 
+#ifdef CONFIG_SMP
 	if (timer->is_soft && base != &migration_base) {
+#else
+	if (timer->is_soft && base && base->cpu_base) {
+#endif
 		spin_lock(&base->cpu_base->softirq_expiry_lock);
 		spin_unlock(&base->cpu_base->softirq_expiry_lock);
 	}
-- 
2.23.0.rc1

