From eb0fddd70c1ef73864f0e71cc12f2a40bc6d40d3 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 25 May 2026 18:19:18 -0500
Subject: [PATCH 3/3] [nfm] quiet ASoC by default

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 sound/soc/soc-utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/soc-utils.c b/sound/soc/soc-utils.c
index 9cb7567e263e..cf33fcafb5bc 100644
--- a/sound/soc/soc-utils.c
+++ b/sound/soc/soc-utils.c
@@ -35,7 +35,7 @@ int snd_soc_ret(const struct device *dev, int ret, const char *fmt, ...)
 		vaf.fmt = fmt;
 		vaf.va = &args;
 
-		dev_err(dev, "ASoC error (%d): %pV", ret, &vaf);
+		dev_dbg(dev, "ASoC error (%d): %pV", ret, &vaf);
 		va_end(args);
 	}
 
-- 
2.53.0

