From 12afd38813e9708e42d4f6ee40402bfd01592bc1 Mon Sep 17 00:00:00 2001
From: Vaishnav M A <vaishnav@beagleboard.org>
Date: Thu, 17 Dec 2020 16:07:45 +0530
Subject: [PATCH] mikrobus_add check for missing w1-gpio

Signed-off-by: Vaishnav M A <vaishnav@beagleboard.org>
---
 drivers/misc/mikrobus/mikrobus_core.c | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/drivers/misc/mikrobus/mikrobus_core.c b/drivers/misc/mikrobus/mikrobus_core.c
index 07f9f3cc9991..f736df3a89fc 100644
--- a/drivers/misc/mikrobus/mikrobus_core.c
+++ b/drivers/misc/mikrobus/mikrobus_core.c
@@ -668,11 +668,20 @@ static int mikrobus_port_id_eeprom_probe(struct mikrobus_port *port)
 	platform_device_register(&mikrobus_id_eeprom_w1_device);
 	port->w1_gpio = &mikrobus_id_eeprom_w1_device;
 	bm = (struct w1_bus_master *) platform_get_drvdata(&mikrobus_id_eeprom_w1_device);
-	port->w1_master = w1_find_master_device(bm);
-	mutex_lock(&port->w1_master->mutex);
-	port->w1_master->max_slave_count = 1;
-	clear_bit(W1_WARN_MAX_COUNT, &port->w1_master->flags);
-	mutex_unlock(&port->w1_master->mutex);
+	if(bm) {
+		port->w1_master = w1_find_master_device(bm);
+		if(!port->w1_master){
+			dev_err(&port->dev, "failed to find W1 GPIO master, port [%s]\n",
+									port->name);
+			gpiod_remove_lookup_table(lookup);
+			kfree(lookup);
+			return -ENODEV;
+		}
+		mutex_lock(&port->w1_master->mutex);
+		port->w1_master->max_slave_count = 1;
+		clear_bit(W1_WARN_MAX_COUNT, &port->w1_master->flags);
+		mutex_unlock(&port->w1_master->mutex);
+	}
 	return 0;
 }
 
-- 
2.25.1

