From 1a49e6babe519ccf5a3c113296d3e2fb7f88b4af Mon Sep 17 00:00:00 2001
From: Mark Brown <broonie@linaro.org>
Date: Fri, 7 Mar 2014 11:43:04 +0800
Subject: [PATCH 06/20] ASoC: pcm512x: Fix duplicate const warning

Signed-off-by: Mark Brown <broonie@linaro.org>
---
 sound/soc/codecs/pcm512x.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/codecs/pcm512x.c b/sound/soc/codecs/pcm512x.c
index 0c90705..4b4c0c7 100644
--- a/sound/soc/codecs/pcm512x.c
+++ b/sound/soc/codecs/pcm512x.c
@@ -201,10 +201,10 @@ static const unsigned int pcm512x_dsp_program_values[] = {
 	7,
 };
 
-static const SOC_VALUE_ENUM_SINGLE_DECL(pcm512x_dsp_program,
-					PCM512x_DSP_PROGRAM, 0, 0x1f,
-					pcm512x_dsp_program_texts,
-					pcm512x_dsp_program_values);
+static SOC_VALUE_ENUM_SINGLE_DECL(pcm512x_dsp_program,
+				  PCM512x_DSP_PROGRAM, 0, 0x1f,
+				  pcm512x_dsp_program_texts,
+				  pcm512x_dsp_program_values);
 
 static const char * const pcm512x_clk_missing_text[] = {
 	"1s", "2s", "3s", "4s", "5s", "6s", "7s", "8s"
-- 
2.1.4

