From a9bbe1aed7a516ea0945ac9eddd78f9f7692f84d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 12 Aug 2014 10:22:46 -0500
Subject: [PATCH] dtb: sync Makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index adb5ed9..4dcb90e 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -288,7 +288,14 @@ dtb-$(CONFIG_ARCH_OMAP3) += am3517-craneboard.dtb \
 	omap3-zoom3.dtb
 dtb-$(CONFIG_SOC_AM33XX) += am335x-base0033.dtb \
 	am335x-bone.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO4.dtb \
+	am335x-bone-ttyO5.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO4.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-nano.dtb
-- 
2.0.1

