From ece27ab155f18980fa13532518fa6356796aa27d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 27 Oct 2021 12:04:31 -0500
Subject: [PATCH] debian: HOST: use max of gcc-9/g++-9

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f81ad24eac60..b4ca06dcaefa 100644
--- a/Makefile
+++ b/Makefile
@@ -239,8 +239,8 @@ CONFIG_SHELL := $(shell if [ -x "$$BASH" ]; then echo $$BASH; \
 	  else if [ -x /bin/bash ]; then echo /bin/bash; \
 	  else echo sh; fi ; fi)
 
-HOSTCC       = gcc
-HOSTCXX      = g++
+HOSTCC       = gcc-9
+HOSTCXX      = g++-9
 HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer -std=gnu89
 HOSTCXXFLAGS = -O2
 
-- 
2.30.2

