From 3ae4e2a82a1efee2a37b50696341f597979fa149 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 24 Sep 2018 15:00:03 -0500
Subject: [PATCH 40/40] fix: include/linux/of.h add linux/slab.h include

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/linux/of.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/of.h b/include/linux/of.h
index 858ca64580a2..2cab25d560ed 100644
--- a/include/linux/of.h
+++ b/include/linux/of.h
@@ -26,6 +26,7 @@
 #include <linux/property.h>
 #include <linux/list.h>
 #include <linux/rhashtable.h>
+#include <linux/slab.h>
 
 #include <asm/byteorder.h>
 #include <asm/errno.h>
-- 
2.19.0

