From f42db1bf9f9751db91a5a7b8e0128f2a7d18bcc7 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 7 Dec 2021 18:55:55 -0600
Subject: [PATCH 6/6] merge: aufs-rt

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 fs/aufs/i_op.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/aufs/i_op.c b/fs/aufs/i_op.c
index bfce27d451f1..4accad3d8b26 100644
--- a/fs/aufs/i_op.c
+++ b/fs/aufs/i_op.c
@@ -637,7 +637,11 @@ int au_pin_hdir_relock(struct au_pin *p)
 
 static void au_pin_hdir_set_owner(struct au_pin *p, struct task_struct *task)
 {
+#if IS_ENABLED(CONFIG_PREEMPT_RT)
+	p->hdir->hi_inode->i_rwsem.rtmutex.owner = task;
+#else
 	atomic_long_set(&p->hdir->hi_inode->i_rwsem.owner, (long)task);
+#endif
 }
 
 void au_pin_hdir_acquire_nest(struct au_pin *p)
-- 
2.30.2

