From 0f22888ea9edb80e4241aedb074e8e49f40b39dc Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 18 Mar 2022 11:54:34 -0500
Subject: [PATCH 6/6] merge: aufs-rt

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 fs/aufs/i_op.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/aufs/i_op.c b/fs/aufs/i_op.c
index dd59340637cf..9fe18fc0f239 100644
--- a/fs/aufs/i_op.c
+++ b/fs/aufs/i_op.c
@@ -636,7 +636,11 @@ int au_pin_hdir_relock(struct au_pin *p)
 
 static void au_pin_hdir_set_owner(struct au_pin *p, struct task_struct *task)
 {
+#if IS_ENABLED(CONFIG_PREEMPT_RT)
+	p->hdir->hi_inode->i_rwsem.rtmutex.owner = task;
+#else
 	atomic_long_set(&p->hdir->hi_inode->i_rwsem.owner, (long)task);
+#endif
 }
 
 void au_pin_hdir_acquire_nest(struct au_pin *p)
-- 
2.30.2

