From 45e1279deadf528007ebb48a9fd48106749571fb Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 4 Sep 2025 15:48:09 -0500
Subject: [PATCH] backports: tps65219: from: linux.git

Reference: v6.7.12
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mfd/tps65219.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mfd/tps65219.c b/drivers/mfd/tps65219.c
index 72a5f51fe32a..0e0c42e4fdfc 100644
--- a/drivers/mfd/tps65219.c
+++ b/drivers/mfd/tps65219.c
@@ -228,6 +228,7 @@ static struct regmap_irq_chip tps65219_irq_chip = {
 static int tps65219_probe(struct i2c_client *client)
 {
 	struct tps65219 *tps;
+	unsigned int chipid;
 	bool pwr_button;
 	int ret;
 
@@ -252,6 +253,12 @@ static int tps65219_probe(struct i2c_client *client)
 	if (ret)
 		return ret;
 
+	ret = regmap_read(tps->regmap, TPS65219_REG_TI_DEV_ID, &chipid);
+	if (ret) {
+		dev_err(tps->dev, "Failed to read device ID: %d\n", ret);
+		return ret;
+	}
+
 	ret = devm_mfd_add_devices(tps->dev, PLATFORM_DEVID_AUTO,
 				   tps65219_cells, ARRAY_SIZE(tps65219_cells),
 				   NULL, 0, regmap_irq_get_domain(tps->irq_data));
-- 
2.51.0

