From 347d00edc0330bf464dee791b4e91acab522e68f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 8 Jul 2024 12:35:33 -0500
Subject: [PATCH 1/2] [TEST]: EMMC04G-MK27 add MMC_QUIRK_TRIM_BROKEN like
 EMMC04G-M627

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/core/quirks.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index ddefedc9746e..7c72807c3e23 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -136,6 +136,14 @@ static const struct mmc_fixup __maybe_unused mmc_blk_fixups[] = {
 	MMC_FIXUP("M62704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
 		  MMC_QUIRK_TRIM_BROKEN),
 
+	/*
+	 * Kingston EMMC04G-MK27 are showing fast wear rates
+	 * 2024???? : testing MMC_QUIRK_TRIM_BROKEN (eMMC died)
+	 * 20240708 : testing MMC_QUIRK_BROKEN_SD_DISCARD (testing)
+	 */
+	MMC_FIXUP("MK2704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
+		  MMC_QUIRK_BROKEN_SD_DISCARD),
+
 	/*
 	 * Some SD cards reports discard support while they don't
 	 */
-- 
2.39.2

