From ebc206f9714d01ae5502391b17f7f1bfe7f752d9 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 16 Apr 2024 10:43:06 -0500
Subject: [PATCH 1/2] [TEST]: EMMC04G-MK27 add MMC_QUIRK_TRIM_BROKEN like
 EMMC04G-M627

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/core/quirks.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index a7ffbc930ea9..4189d715c038 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -109,6 +109,13 @@ static const struct mmc_fixup __maybe_unused mmc_blk_fixups[] = {
 	MMC_FIXUP("V10016", CID_MANFID_KINGSTON, CID_OEMID_ANY, add_quirk_mmc,
 		  MMC_QUIRK_TRIM_BROKEN),
 
+	/*
+	 * Kingston EMMC04G-MK27 are showing fast wear rates, lets disable
+	 * TRIM like EMMC04G-M627 and see if that helps, on fresh boards
+	 */
+	MMC_FIXUP("MK2704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
+		  MMC_QUIRK_TRIM_BROKEN),
+
 	/*
 	 * Some SD cards reports discard support while they don't
 	 */
-- 
2.39.2

