From b3271b271cf80e0e1d5e9726dd10d550dc054a33 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 16 Apr 2024 10:44:29 -0500
Subject: [PATCH 2/2] [debug]: mmc: quirks more vocal

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/core/quirks.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index 23a78aeb18f8..f14a41b0f0e5 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -245,7 +245,7 @@ static inline void mmc_fixup_device(struct mmc_card *card,
 		if (f->month != CID_MONTH_ANY && f->month != card->cid.month)
 			continue;
 
-		dev_dbg(&card->dev, "calling %ps\n", f->vendor_fixup);
+		dev_info(&card->dev, "calling %ps\n", f->vendor_fixup);
 		f->vendor_fixup(card, f->data);
 	}
 }
-- 
2.39.2

