From e8b6527f27069d97fb3b3dfa18595d43efe7385c Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 19 Nov 2014 11:13:43 -0600
Subject: [PATCH 3/3] deb-pkg: no dtbs_install

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/package/builddeb | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 5083ad9..f32d59b 100644
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -167,7 +167,8 @@ fi
 
 if grep -q '^CONFIG_OF=y' $KCONFIG_CONFIG ; then
 	mkdir -p "$tmpdir/boot/dtbs/$version"
-	INSTALL_DTBS_PATH="$tmpdir/boot/dtbs/$version" $MAKE KBUILD_SRC= dtbs_install
+	$MAKE KBUILD_SRC= dtbs
+	find arch/arm/boot/ -iname "*.dtb" -exec cp -v '{}' "$tmpdir/boot/dtbs/$version" \;
 
 	#make dtbs_install seems to add an .old directory
 	if [ -d "$tmpdir/boot/dtbs/$version.old" ] ; then
-- 
2.1.4

