From 59d4fe2d97f8e0845385465c8c64a780ed958181 Mon Sep 17 00:00:00 2001
From: Frank Binns <frank.binns@imgtec.com>
Date: Sun, 19 Dec 2021 19:13:42 +0000
Subject: [PATCH 5/5] MAINTAINERS: Add entry for PowerVR DRM driver

Signed-off-by: Frank Binns <frank.binns@imgtec.com>
---
 MAINTAINERS | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 05fd080b82f3..c754e9575f6a 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -9380,6 +9380,15 @@ IMGTEC IR DECODER DRIVER
 S:	Orphan
 F:	drivers/media/rc/img-ir/
 
+IMGTEC POWERVR DRM DRIVER
+M:	Frank Binns <frank.binns@imgtec.com>
+M:	Sarah Walker <sarah.walker@imgtec.com>
+R:	Matt Coster <matt.coster@imgtec.com>
+S:	Supported
+F:	Documentation/gpu/imagination/
+F:	drivers/gpu/drm/imagination/
+F:	include/uapi/drm/pvr_drm.h
+
 IMON SOUNDGRAPH USB IR RECEIVER
 M:	Sean Young <sean@mess.org>
 L:	linux-media@vger.kernel.org
-- 
2.30.2

