From 8ad6e4fb0b088aae5a3d5f4a5c85f69af7272a67 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 25 Feb 2015 13:11:24 -0600
Subject: [PATCH 2/2] rt: we append rt on our own

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 localversion-rt | 1 -
 1 file changed, 1 deletion(-)
 delete mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
deleted file mode 100644
index 5a64c9c..0000000
--- a/localversion-rt
+++ /dev/null
@@ -1 +0,0 @@
--rt5
-- 
2.1.4

