From 657cdf52dc4079e4c72a0a2edf4d5f274d483de8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 29 Dec 2017 13:53:31 -0600
Subject: [PATCH 2/2] rt: fix missing types.h in spinlock_types.h

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/include/asm/spinlock_types.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/include/asm/spinlock_types.h b/arch/arm/include/asm/spinlock_types.h
index a37c0803954b..c58af616d54e 100644
--- a/arch/arm/include/asm/spinlock_types.h
+++ b/arch/arm/include/asm/spinlock_types.h
@@ -2,6 +2,8 @@
 #ifndef __ASM_SPINLOCK_TYPES_H
 #define __ASM_SPINLOCK_TYPES_H
 
+#include <linux/types.h>
+
 #define TICKET_SHIFT	16
 
 typedef struct {
-- 
2.15.1

