From 6abeac2224e8c389272c2ee4284d6bd7748b34ec Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 28 Aug 2019 16:52:55 -0500
Subject: [PATCH 2/2] greybus: drivers/staging/greybus/module.c: no struct_size
 yet

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/staging/greybus/module.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/greybus/module.c b/drivers/staging/greybus/module.c
index 894d02e8d8b7..b785382192de 100644
--- a/drivers/staging/greybus/module.c
+++ b/drivers/staging/greybus/module.c
@@ -94,8 +94,8 @@ struct gb_module *gb_module_create(struct gb_host_device *hd, u8 module_id,
 	struct gb_module *module;
 	int i;
 
-	module = kzalloc(struct_size(module, interfaces, num_interfaces),
-			 GFP_KERNEL);
+	module = kzalloc(sizeof(*module) + num_interfaces * sizeof(intf),
+				GFP_KERNEL);
 	if (!module)
 		return NULL;
 
-- 
2.23.0.rc1

