From 31fba5dd3fa2ad4a17232ffe876ba84bb878f28f Mon Sep 17 00:00:00 2001
From: Michael Haberler <git@mah.priv.at>
Date: Fri, 12 Jul 2013 21:07:21 +0200
Subject: [PATCH 2/3] xenomai: apply Stephan Kappertz's post.patch

---
 arch/arm/mach-omap2/timer.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/timer.c b/arch/arm/mach-omap2/timer.c
index 9dc10d2a..22ed3e3 100644
--- a/arch/arm/mach-omap2/timer.c
+++ b/arch/arm/mach-omap2/timer.c
@@ -430,7 +430,7 @@ static void __init omap2_gp_clockevent_init(int gptimer_id,
 
 		ipipe_timer_register(&omap3_itimer);
 	}
-	if (cpu_is_omap44xx() && num_possible_cpus() == 1)
+	if ((cpu_is_omap44xx() || soc_is_am33xx()) && num_possible_cpus() == 1)
 		ipipe = 1;
 #endif /* CONFIG_IPIPE */
 
@@ -455,7 +455,7 @@ static void __init omap2_gp_clockevent_init(int gptimer_id,
 	clockevent_gpt.irq = omap_dm_timer_get_irq(&clkev);
 
 #ifdef CONFIG_IPIPE
-	if (cpu_is_omap44xx() && num_possible_cpus() == 1) {
+	if ((cpu_is_omap44xx() || soc_is_am33xx()) && num_possible_cpus() == 1) {
 		omap4_itimer.irq = clkev.irq;
 		omap4_itimer.min_delay_ticks = 3;
 
@@ -570,6 +570,7 @@ static void __init omap2_gptimer_clocksource_init(int gptimer_id,
 						const char *fck_source)
 {
 	int res, ipipe = 0;
+	int timer_func_offset;
 
 #ifdef CONFIG_IPIPE
 	ipipe = 1;
@@ -594,11 +595,12 @@ static void __init omap2_gptimer_clocksource_init(int gptimer_id,
 
 #if defined(CONFIG_IPIPE)
 	if (ipipe) {
+		timer_func_offset = (clksrc.revision == 2) ? OMAP_TIMER_V2_FUNC_OFFSET : 0;
 		tsc_info.freq = clksrc.rate;
 		tsc_info.counter_vaddr =
-			(unsigned long)clksrc.io_base + (OMAP_TIMER_COUNTER_REG & 0xff);
+			(unsigned long)clksrc.io_base + timer_func_offset + (OMAP_TIMER_COUNTER_REG & 0xff);
 		tsc_info.u.counter_paddr =
-			clksrc.phys_base + (OMAP_TIMER_COUNTER_REG & 0xff);
+			clksrc.phys_base + timer_func_offset + (OMAP_TIMER_COUNTER_REG & 0xff);
 		__ipipe_tsc_register(&tsc_info);
 	}
 #endif
-- 
1.7.10.4

