From 7e8b2d970f4fa646051d4656949f4d9c80e79a1c Mon Sep 17 00:00:00 2001
From: Michael Haberler <git@mah.priv.at>
Date: Tue, 21 Apr 2015 19:08:29 +0200
Subject: [PATCH 1/1] config: add RT-CAN options for Bosch CCAN,DCAN drivers

---
 drivers/xenomai/can/Kconfig |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/xenomai/can/Kconfig b/drivers/xenomai/can/Kconfig
index 62f542b..574f302 100644
--- a/drivers/xenomai/can/Kconfig
+++ b/drivers/xenomai/can/Kconfig
@@ -95,5 +95,6 @@ config XENO_DRIVERS_CAN_CCAN
 
 source drivers/xenomai/can/mscan/Kconfig
 source drivers/xenomai/can/sja1000/Kconfig
+source drivers/xenomai/can/c_can/Kconfig
 
 endmenu
-- 
1.7.10.4

