diff -uPr ipipe/arch/arm/mach-omap2/gpmc.c post/arch/arm/mach-omap2/gpmc.c
--- ipipe/arch/arm/mach-omap2/gpmc.c	2013-08-27 20:34:22.366199351 +0200
+++ post/arch/arm/mach-omap2/gpmc.c	2013-08-27 20:16:56.990703412 +0200
@@ -25,6 +25,12 @@
 #include <linux/module.h>
 #include <linux/interrupt.h>
 #include <linux/platform_device.h>
+#include <linux/of.h>
+#include <linux/of_address.h>
+#include <linux/of_mtd.h>
+#include <linux/of_device.h>
+#include <linux/mtd/nand.h>
+#include <linux/pinctrl/consumer.h>
 #include <linux/ipipe.h>
 
 #include <linux/platform_data/mtd-nand-omap2.h>
