diff -uPr bb/arch/arm/mach-omap2/gpmc.c pre/arch/arm/mach-omap2/gpmc.c
--- bb/arch/arm/mach-omap2/gpmc.c	2013-08-27 20:26:55.080672567 +0200
+++ pre/arch/arm/mach-omap2/gpmc.c	2013-08-27 20:31:15.727578153 +0200
@@ -25,12 +25,6 @@
 #include <linux/module.h>
 #include <linux/interrupt.h>
 #include <linux/platform_device.h>
-#include <linux/of.h>
-#include <linux/of_address.h>
-#include <linux/of_mtd.h>
-#include <linux/of_device.h>
-#include <linux/mtd/nand.h>
-#include <linux/pinctrl/consumer.h>
 
 #include <linux/platform_data/mtd-nand-omap2.h>
 
