From 4b22c8aa41bd3a3b88a5e8ab17b278dcb19afb54 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 10 Jun 2020 13:32:57 -0500
Subject: [PATCH 2/2] WireGuard fixes ipv6_dst_lookup_flow was backported

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 net/wireguard/compat/compat.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/net/wireguard/compat/compat.h b/net/wireguard/compat/compat.h
index 824f57cad01f2..65dcb8793b5e1 100644
--- a/net/wireguard/compat/compat.h
+++ b/net/wireguard/compat/compat.h
@@ -861,10 +861,6 @@ static inline void skb_mark_not_on_list(struct sk_buff *skb)
 })
 #endif
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 5) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)) || LINUX_VERSION_CODE < KERNEL_VERSION(5, 3, 18)
-#define ipv6_dst_lookup_flow(a, b, c, d) ipv6_dst_lookup(a, b, &dst, c) + (void *)0 ?: dst
-#endif
-
 #if defined(ISUBUNTU1604)
 #include <linux/siphash.h>
 #ifndef _WG_LINUX_SIPHASH_H
-- 
2.26.2

