From f2eb29ae38906c1e04c9621b51f8b0c8867c05b6 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 8 Mar 2021 15:01:17 -0600
Subject: [PATCH 3/3] wireguard, skb_mark_not_on_list was moved into 4.14.217

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 net/wireguard/compat/compat.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireguard/compat/compat.h b/net/wireguard/compat/compat.h
index 65dcb8793b5e..99b690402b12 100644
--- a/net/wireguard/compat/compat.h
+++ b/net/wireguard/compat/compat.h
@@ -824,7 +824,7 @@ static __always_inline void old_rcu_barrier(void)
 #define COMPAT_CANNOT_DEPRECIATE_BH_RCU
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 10) && !defined(ISRHEL8)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 14, 217) && !defined(ISRHEL8)
 static inline void skb_mark_not_on_list(struct sk_buff *skb)
 {
 	skb->next = NULL;
-- 
2.30.1

