From b92fea206d414a9cf8b024a6a6aea3746ffb2d7a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 8 Mar 2021 15:03:26 -0600
Subject: [PATCH 4/4] wireguard: skb_list_walk_safe was merged in v4.14.217

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 net/wireguard/device.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/net/wireguard/device.h b/net/wireguard/device.h
index c91f3051c5c7..b15a8be9d816 100644
--- a/net/wireguard/device.h
+++ b/net/wireguard/device.h
@@ -62,12 +62,4 @@ struct wg_device {
 int wg_device_init(void);
 void wg_device_uninit(void);
 
-/* Later after the dust settles, this can be moved into include/linux/skbuff.h,
- * where virtually all code that deals with GSO segs can benefit, around ~30
- * drivers as of writing.
- */
-#define skb_list_walk_safe(first, skb, next)                                   \
-	for (skb = first, next = skb->next; skb;                               \
-	     skb = next, next = skb ? skb->next : NULL)
-
 #endif /* _WG_DEVICE_H */
-- 
2.30.1

